external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2280,6 +2545,202 @@ fn parse_parameter(context: &mut Context) -> Result<(Mutability, Var, Type), Box
     Ok((mut_, v, t))
 }
 
+//**************************************************************************************************
+// Enums
+//**************************************************************************************************
+
+// Parse an enum definition:
+//      EnumDecl =
+//          "enum" <EnumDefName> ("has" <Ability> (, <Ability>)+)?
+//          "{" (<VariantDecl>,)+ "}" ("has" <Ability> (, <Ability>)+;)
+//      EnumDefName =
+//          <Identifier> <OptionalTypeParameters>
+// Where the the two "has" statements are mutually exclusive -- an enum cannot be declared with
+// both infix and postfix ability declarations.
+
+fn parse_enum_decl(
+    attributes: Vec<Attributes>,
+    start_loc: usize,
+    modifiers: Modifiers,
+    context: &mut Context,
+) -> Result<EnumDefinition, Box<Diagnostic>> {
+    let Modifiers {
+        visibility,
+        entry,
+        native: _,
```

Comment:
I think we need check also check for a native specifier and error on native as well here.

E.g., I think the following is currently permitted and shouldn't be:
```
public native enum E { V() }
```